src/include/sof/audio/audio_stream.h

```diff
@@ -0,0 +1,259 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
```

Comment:
@ktrzcinx there are some inconsistencies in the style. Some functions are starting with stream some others with audio_stream.

Please use one and stick with it. I think it make sense for everthing to start with stream since we are already in audio/ directory. Including the file should be named audio/stream.h instead of audio/audio_stream.h

Also struct audio_stream should go as struct stream .

Reply:
Yea, I also thought for a while about name. I thought about 'stream' but there are file ipc/stream.h so  I added audio_ prefix to make it more unique, but audio_stream_function_name() lead to very long function names, so here I reject audio_prefix. 

Maybe 'astream' or 'audios' are better solutions? In terms of uniqueness and number of letters in prefix. Or maybe some acronym? For example 'cas' - circular audio stream - it will be unique and short, less self-descriptive, but it will be common structure, so maybe?

Reply:
@ktrzcinx Indeed naming is hard. We should also have some conventions like kernel has for internal/external APIs.

Reply:
@dbaluta by "special type of audio buffer, instead of using a generic buffer" you mean that it's not just a circular buffer but have extra fields like audio format, channels and rate ? If yes, then I can say that we are making audio firmware :P so it is our priority to have user-friendly api for audio processing. Moreover byte-stream without information about format is some kind of black box.

And what would you like to choose in naming issue? Maybe I stick to structure name 'audio_stream' but change functions prefix to 'as_'? For me it sound quite descriptive and short in typing? But I would like to here some approval of this concept or another proposition before changes.

Reply:
my understanding now (by mentally squashing patch 2 & 3) is that this is generic buffer code.

Reply:
i dont see as_ prefix to be very useful. If there is no better option just use audio_stream even if the lines are larger.

Reply:
@dbaluta good point - @ktrzcinx can you update.